### PR TITLE
Canceling stream context only in case of error

### PR DESCRIPTION
**Pull request by thiagodpf**
I imagine there's no reason to cancel the stream context unless there's actually an error.

Related to https://github.com/grafana/xk6-grpc/issues/29

### CODE REVIEW

File: grpc/stream.go [hunk 1]
```diff
@@ -343,12 +344,16 @@ func (s *stream) close(err error) {
 	s.logger.WithError(err).Debug("stream is closing")
 
 	s.state = closed
-	close(s.done)
+
+	s.doneOnce.Do(func() {
+		close(s.done)
+	})
```
Comment (mstoykov) [inline grpc/stream.go:350]: This should never be needed as this function should only be called on the event loop so synchronized and the check above should prevent any race.

Did you experience a race?

Comment (thiagodpf) [reply grpc/stream.go:350]: You're absolutely right, my fault! For a moment I forgot about the javascript event loop. I'm sorry, I'll undo this!

**Comment by olegbespalov** [comment]
Hey @thiagodpf !

Thanks again for the contribution and for actually pointing out the issue. I have a question, does this fix solve your issue?

I'm asking because from the top of my head, I still see the issue with the bi-directional streams since we still do `close(s.done)` that they will stop receiving messages :cry:

**Comment by thiagodpf** [comment]
Hi @olegbespalov !

You're right, I think I biased my analysis because I was debugging k6 and my application at the same time without taking into account the independence between the two programs, I really owed a test scenario in this PR. 

At that moment, my intention was to stop the premature cancellation of the context and I ended up not realizing that the structure that encapsulates the stream was also being closed. I think it's better to cancel this PR so as not to pollute your issues board. :sweat_smile:

Unfortunately I lack knowledge of the code to be able to propose a decent solution, my fault. :grin:

**Comment by olegbespalov** [comment]
No worries at all! I'll try to prioritize that and suggest a fix in an upcoming cycle. Anyway, thanks for your input!

**Pull request closed without merging**